### ISSUE TITLE
Kernel panic on kubernetes nodes, ipt_do_table+0x60f/0x640 general protection fault

### ISSUE THREAD

**Issue by ruslanloman** [projectcalico/felix#2655]
Hi, 
We run kubernetes cluster on top of OpenStack with about 400 k8s worker nodes. After our migration from kube-router to calico we started to observe random kernel panic on worker nodes. We have multiple clusters with the same hardware, bios configuration/firmware, only major difference is kube-router used there and it works without problem.

## Expected Behavior
System works without issue.

## Current Behavior
Random kernel panic on nodes with running calico.
kernel panic log:  [14-Jan-2021.log](https://github.com/projectcalico/felix/files/5821079/14-Jan-2021.log)

## Possible Solution
Not clear, we tried different configuration options, update to latest stable kernel version provided by flatcar - didn't help.

## Steps to Reproduce (for bugs)
Unfortunately it's not clear how to reproduce. It happens randomly, several times a day

## Your Environment
We couldn't upgrade to the latest stable calico version since it requires newer kubernetes. 
* `calico/node` container version (if running Felix in a container): calico/node:v3.13.5
* Which orchestrator are you using (e.g. Kubernetes, OpenStack, Docker, Mesos): Kubernetes
* Orchestrator version (if used): v1.15.3
* Operating System and version: Issue appears on flatcar as well as on deprecated coreos version

Flatcar:
```
kernel: 5.4.83-flatcar
kernel params: rootflags=rw mount.usrflags=ro BOOT_IMAGE=/flatcar/vmlinuz-b mount.usr=/dev/mapper/usr verity.usr=PARTUUID=e03dd35c-7c2d-4a47-b3fe-27f15780a57c rootflags=rw mount.usrflags=ro consoleblank=0 root=LABEL=ROOT console=ttyS0,115200n8 console=tty0 flatcar.oem.id=openstack verity.usrhash=864b943d059231f1d1c90f2dccbc43fcf494ed1e2ae9750f411bf9963ca74227
DISTRIB_ID="Flatcar Container Linux by Kinvolk"
DISTRIB_RELEASE=2605.10.0
```
CoreOS:
```
kernel: 4.19.68-coreos
kernel params: rootflags=rw mount.usrflags=ro BOOT_IMAGE=/coreos/vmlinuz-a mount.usr=/dev/mapper/usr verity.usr=PARTUUID=7130c94a-213a-4e5a-8e26-6cce9662f132 rootflags=rw mount.usrflags=ro consoleblank=0 root=LABEL=ROOT console=ttyS0,115200n8 console=tty0 coreos.oem.id=openstack crashkernel=256M verity.usrhash=b2e73c3a2f18b1795d6af3668efa591c87d371483402122acd9611df539c3238
DISTRIB_ID="Container Linux by CoreOS"
DISTRIB_RELEASE=2191.5.0
```
* Calico config:  [calico-config.log](https://github.com/projectcalico/felix/files/5821057/calico-config.log)
* Calico env variables: [calico-venv.log](https://github.com/projectcalico/felix/files/5821151/calico-venv.log)

**Comment by neiljerram** [comment]
@ruslanloman The only thing that springs to mind is iptables legacy vs NFT mode.  Perhaps the modes are being mixed, on the node as a whole, and that is causing the panic.  Felix in v3.13.5 defaulted to using legacy mode, so perhaps it will help to set FELIX_IPTABLESBACKEND to "nft" instead.

**Comment by caseydavenport** [comment]
@ruslanloman did you have any luck trying to update Calico's config t ouse FELIX_IPTABLESBACKEND=nft as suggested? Seems like a potential culprit. On the same note, could you confirm the iptables version / mode in-use on the host?

**Comment by ruslanloman** [comment]
Hi @neiljerram @caseydavenport,
  Thanks for your advice with FELIX_IPTABLESBACKEND=nft, we tried it on testing and it's caused downtime for our workload, so we decided to enable kdump and tried to analyze kernel dump. Seems we found the root cause and managed to mitigate it. We use kubespray to deploy k8s and it installs nodelocaldns [1] service on each worker node to cache dns requests. This service also manages iptables rules and has default refresh interval is 60s [2](felix one is 90s [3]), in kernel dump there was same ip address all the time that belongs to virtual interface of  nodelocaldns service. We increased  nodelocaldns iptables refresh interval to 2h and no more kernel panics so far. I think using iptables nft backend for felix will fix issue as well.

[1] - https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/
[2] - https://github.com/kubernetes/dns/blob/master/cmd/node-cache/main.go#L69
[3] - https://docs.projectcalico.org/reference/felix/configuration
